pkg/pgxconn/pgx_conn.go

```diff
@@ -37,21 +47,63 @@ type connImpl struct {
 	Conn *pgxpool.Pool
 }
 
+type pgxRows struct {
+	pgx.Rows
+	sqlQuery  string
+	args      []interface{}
+	startTime time.Time
+	loggged   bool
+}
+
 func (p *connImpl) Close() {
 	conn := p.Conn
 	p.Conn = nil
 	conn.Close()
 }
 
-func (p *connImpl) Exec(ctx context.Context, sql string, arguments ...interface{}) (pgconn.CommandTag, error) {
-	return p.Conn.Exec(ctx, sql, arguments...)
+func (p *pgxRows) Next() bool {
+	if !p.loggged {
+		p.loggged = true
+		LogQueryStats(p.sqlQuery, p.startTime, p.args)()
+	}
+	return p.Rows.Next()
+}
+
+func (p *pgxRows) Scan(dest ...interface{}) error {
+	return p.Rows.Scan(dest...)
+}
+
+func (p *pgxRows) Err() error {
+	return p.Rows.Err()
+}
+
+func (p *pgxRows) Close() {
+	p.Rows.Close()
+}
+
+// calc SQL query execution time
+func LogQueryStats(sql string, startTime time.Time, args ...interface{}) func() {
+	if startTime.IsZero() {
+		startTime = time.Now()
+	}
+	return func() {
+		log.Debug("msg", fmt.Sprintf("time taken by SQL query: %s with args: %v is %v", sql, args, time.Since(startTime)))
+	}
+}
+
+func (p *connImpl) Exec(ctx context.Context, sql string, args ...interface{}) (pgconn.CommandTag, error) {
+	defer LogQueryStats(sql, time.Time{}, args...)()
```

Comment:
I do worry creating this on all connImpl will log the ingest queries as well as the promql queries. I think we probably want just the promql queries logged since otherwise the ingest will just fill the logs. So I'd suggest either having two different connImpl's (one that logs one that doesn't) or add a log bool to connImpl to turn logging on and off